### DISCUSSION TITLE
Conditional sets using sequence

### DISCUSSION THREAD

**Discussion by Ali-Zmn** [data-driven-forms/react-forms#1248] [Need help] [answered]
I have a field which sets its value according to 3 other fields. Initially, it works fine, but after a couple of switches, it looses its functionality.
For example in [this sandbox](https://codesandbox.io/s/react-data-driven-sequence-reported-nmmv6f) if `Provider E` in the first field is chosen, the `some title` field will generate with `key 1` (expected behaviour) then if provider changes to `Provider F` field would regenerate to `key 7` and `key 8` (expected behaviour) **BUT** if provider switches back to `Provider E` the `some title` field won't act as expected and empty itself!

I'm wondering if I'm using the sequence wrong or missing something there

**Comment by rvsia** [comment, marked as answer]
Hello @Ali-Zmn,

the sequence condition works differently >> `This special type of condition allows to trigger a sequence of multiple independent conditions. (...) Setters are executed independently.` [Read more here](https://data-driven-forms.org/schema/condition-sequence#heading-sequence).

That means, all the conditions run at once in your code and because you have `else` statements there, some of unsuccessful condition sets the value of `some title` to an empty array.

I have removed all the `else` statements and now it looks working correctly when switching the provider: https://codesandbox.io/s/react-data-driven-sequence-reported-forked-5b2o45

Please take a look. (Maybe you would need to add one `else: { visible: true }` (but without the setter!) - to show the field, at least one condition in the sequence has to be true to show a field.)

**Reply by Ali-Zmn** [reply]
Thanks @rvsia , my only concern is:
When a user uses an option which **is in the sequence**, it would set the value correctly, but if the user changes it after, to any other option that **is not in the sequence**, the field would keep its value which I don't expect it and I **want to clear** it. one approach can be creating another sequence to act in the opposite way of other sequences and empty the field but is there any other approach?

**Reply by rvsia** [reply]
You could just add a negated condition (for example using `not` [see here](https://data-driven-forms.org/schema/not)) to set the value to empty, if the option **not in sequence** is selected.

schematic code:

```
sequence:
   if a = 1 then x = 1;
   if a = 2 then x = 2;
   if a = 3 then x = 3;
   not: [
            if a = 1;
            if a = 2;
            if a = 3;
         ]
         then x = empty
```

example (when field1 or field2 is not x, then "clear this"):

```jsx
condition: {
  not: {
    or: [
      { when: "field1", is: "x" },
      { when: "field2", is: "x" }
    ]
  },
  then: { set: { field3: "clear this" } },
}
```

Or using `notMatch`

```jsx
condition: {
  or: [
    { when: "field1", is: "x", notMatch: true },
    { when: "field2", is: "x", notMatch: true }
  ],
  then: { set: { field3: "clear this" } }
}
```

**Reply by Ali-Zmn** [reply]
Thanks, it worked as expected!

**Reply by rvsia** [reply]
Glad to hear it!
